File: source/code/python/account_management/create_ldap_dir_req.py
directory = client.directories.create({
    'name': 'Captains',
    'description': 'Famous captains throughout history.',
    'provider': {
        'provider_id': 'ldap',
        'agent': {
            'config': {
                'directory_host': 'ldap.local',
                'directory_port': '666',
                'ssl_required': True,
                'agent_user_dn': 'captain@america.com',
                'agent_user_dn_password': 'americaIStheb3st!!',
                'base_dn': 'dc=example,dc=com',
                'poll_interval': 60,
                'account_config': {
                    'dn_suffix': 'ou=employees',
                    'object_class': 'person',
                    'object_filter': '(cn=finance)',
                    'email_rdn': 'email',
                    'given_name_rdn': 'givenName',
                    'middle_name_rdn': 'middleName',
                    'surname_rdn': 'sn',
                    'username_rdn': 'uid',
                    'password_rdn': 'userPassword',
                },
                'group_config': {
                    'dn_suffix': 'ou=groups',
                    'object_class': 'groupOfUniqueNames',
                    'object_filter': '(ou=*-group)',
                    'name_rdn': 'cn',
                    'description_rdn': 'description',
                    'members_rdn': 'uniqueMember'
                }
            }
        }
    }
})

File: source/code/nodejs/account_management/create_ldap_dir_req.js
var newLdapDirectory = {
  name: 'Captains LDAP Directory',
  description: 'Famous captains throughout history.',
  provider:{
    providerId: 'ldap',
    agent: {
      config : {
        directoryHost : 'ldap.example.com',
        directoryPort : 6362,
        sslRequired : true,
        agentUserDn : 'agentUsername',
        agentUserDnPassword: 'uErnb73bXXsu8z',
        baseDn : 'dc=example,dc=com',
        pollInterval : 60,
        accountConfig : {
          dnSuffix : 'ou=employees',
          objectClass : 'person',
          objectFilter : '(cn=finance)',
          emailRdn : 'email',
          givenNameRdn : 'givenName',
          middleNameRdn : 'middleName',
          surnameRdn : 'sn',
          usernameRdn : 'uid',
          passwordRdn : 'userPassword'
        },
        groupConfig : {
          dnSuffix : 'ou=groups',
          objectClass : 'groupOfUniqueNames',
          objectFilter : '(ou=*-group)',
          nameRdn : 'cn',
          descriptionRdn : 'description',
          membersRdn : 'uniqueMember'
        }
      }
    }
  }
};

client.createDirectory(newLdapDirectory, function(err, directory) {
  console.log(err, directory);
});
